### PR TITLE
Seedlet: Update text and link when there are no published posts

### PR DESCRIPTION
**Pull request by jordesign**
Addressing #3874

This PR changes the content shown when a user has 'Latest Posts' selected for their front page but no public posts. This affects only the logged-in view.

### Updated appearance

Screen Shot 2021-05-18 at 3 14 09 pm

## Testing Instructions

1. Ensure site has no public posts (set them to be draft or remove them).
2. Use the Customizer to set the 'Home Page Settings' to show latest posts
3. Visit the home page of the site.

### CODE REVIEW

**Comment by kwight** [comment]
Great @jordesign – can you include testing instructions please?

**Comment by jordesign** [comment]
Okedoke - have updated with testing instructions - sorry about that.

**Comment by mreishus** [comment]
The translated text used to be in the form `Lorem ipsum <a>dolor sit amet</a>` (tag mixed in with text), but now it's in the form `<a>Lorem ipsum dolor sit amet</a>` (tag strictly surrounding text). Given that, is it more appropriate to take the `<a>` tags out of the translatable text and to move them w/ the other surrounding tag (the `<p>`)?

File: seedlet/template-parts/content/content-none.php [hunk 1]
```diff
@@ -12,24 +12,29 @@
...
 
-			printf(
+			<p><?php _e( 'Your site is set to show the the most recent posts on your homepage - but you don\'t have any Posts published.', 'seedlet' ); ?></p>
+
+			<?php printf(
 				'<p>' . wp_kses(
-					/* translators: 1: link to WP admin new post page. */
-					__( 'Ready to publish your first post? <a href="%1$s">Get started here</a>.', 'seedlet' ),
+					/* translators: 1: link to WP admin new post page. 2: class for anchor */
+					__( '<a href="%1$s" class="%2$s">Add or publish Posts</a>', 'seedlet' ),
```
Comment (scruffian) [inline seedlet/template-parts/content/content-none.php]: If all the text is in a link then I think we can make this much simpler by moving the translation function inside the link.

**Comment by jordesign** [comment]
Thanks @mreishus and @scruffian - I've just pushed a fix which removes the anchor from the translatable text and (I think) improves the overall formatting of that section.

Still more than open to feedback on how it could be cleaner and tidier.

File: seedlet/template-parts/content/content-none.php [hunk 2]
```diff
@@ -12,27 +12,35 @@
...
 		<?php
-		if ( is_home() && current_user_can( 'publish_posts' ) ) :
+		if ( is_home() && current_user_can( 'publish_posts' ) ) : ?>
 
-			printf(
-				'<p>' . wp_kses(
-					/* translators: 1: link to WP admin new post page. */
-					__( 'Ready to publish your first post? <a href="%1$s">Get started here</a>.', 'seedlet' ),
+			<p>
+			<?php _e( 'Your site is set to show the the most recent posts on your homepage - but you don\'t have any Posts published.', 'seedlet' ); ?></p>
```
Comment (scruffian) [inline seedlet/template-parts/content/content-none.php]: Very minor, but you could wrap this string with double quotes and then you wouldn't need the `\` on the single quote.

File: seedlet/template-parts/content/content-none.php [hunk 3]
```diff
@@ -12,27 +12,35 @@
...
+
+			<?php printf(
```
Comment (scruffian) [inline seedlet/template-parts/content/content-none.php]: I think you can remove wp_kses and printf here altogether and just write the button as markup.

Comment (jordesign) [reply seedlet/template-parts/content/content-none.php]: Oh awesome - thank for these tips @scruffian - I'm finding my way around our themes repo for the first time an more than happy for suggestions of the neatest way to do things but still have them fit our style/methods.

**Comment by jordesign** [comment]
After a little snafu with work on the wrong branch (oops) this is updated and ready for review again. Thanks folks!

**Comment by scruffian** [comment]
This is looking really good. The only other thing that would be good to do is to adhere to the WordPress coding standards. We have a pre-commit hook which can take care of that for you - more details here: https://github.com/Automattic/themes/blob/trunk/README.md#coding-standards

**Comment by jordesign** [comment]
Thanks @scruffian  - Have run it through the pre-commit hook now - and all looks good. Most recent commit should be correct.

**Comment by scruffian** [comment]
Great, thank you, I'll get this merged.

**Pull request merged**